oneflow/core/functional/impl/nn_functor.cpp

```diff
@@ -264,6 +264,93 @@ class BatchMatMulFunctor {
   std::shared_ptr<OpExpr> batch_matmul_op_;
 };
 
+class TensorDotFunctor {
+ public:
+  Maybe<Tensor> operator()(const std::shared_ptr<Tensor>& a, const std::shared_ptr<Tensor>& b,
+                           const std::vector<int64_t>& dims_a,
+                           const std::vector<int64_t>& dims_b) const {
+    CHECK_EQ_OR_RETURN(dims_a.size(), dims_b.size())
+        << "dims1 and dims2 must have same size, got " << dims_a.size() << " and " << dims_b.size();
+
+    if (dims_a.empty() && dims_b.empty()) {
+      std::vector<int64_t> shape_sum;
```

Comment:
这个地方可以构造DimVector，然后更新值即可，不需要用std::vector后面又构造一次DimVector：

```suggestion
      DimVector shape_sum(a->shape()->NumAxes() + b->shape()->NumAxes());
      for() {shape_sum[i] = xxx; }
```